rpc/src/custom_rpc_server.rs

```diff
@@ -0,0 +1,335 @@
+    };
+
+    // Deserialize the JSON-RPC request.
+    let mut req = match json_rpc2::from_slice(&data) {
+        Ok(req) => req,
+        Err(_) => {
+            let err = json_rpc2::Error::InvalidRequest {
+                data: "Invalid request!".into(),
+            };
+            let resp: json_rpc2::Response = err.into();
+
+            return Ok(hyper::Response::new(
+                serde_json::to_vec(&resp).unwrap_or_default().into(),
+            ));
+        }
+    };
+
+    // Read the request params.
+    let mut params = match read_params(&mut req) {
+        Ok(params) => params,
+        Err(e) => {
+            let resp: json_rpc2::Response = (&mut req, e).into();
+            let body = serde_json::to_vec(&resp).unwrap_or_default();
+
+            return Ok(hyper::Response::new(body.into()));
+        }
+    };
+
+    // Handle the request method.
+    let value = match req.method() {
+        // public
+        "getblock" => {
+            let resp = rpc
+                .get_block(params[0].as_str().unwrap_or("").into())
+                .map_err(convert_crate_err);
+            serde_json::to_value(resp).unwrap_or_default()
+        }
+        "getblockcount" => {
+            let resp = rpc.get_block_count().map_err(convert_crate_err);
+            serde_json::to_value(resp).unwrap_or_default()
+        }
+        "getbestblockhash" => {
+            let resp = rpc.get_best_block_hash().map_err(convert_crate_err);
+            serde_json::to_value(resp).unwrap_or_default()
+        }
+        "getblockhash" => match serde_json::from_value::<u32>(params.remove(0)) {
+            Ok(height) => {
+                let resp = rpc.get_block_hash(height).map_err(convert_crate_err);
+                serde_json::to_value(resp).unwrap_or_default()
+            }
+            Err(_) => {
+                let err = json_rpc2::RpcError {
+                    code: PARSE_ERROR, // as per JSON-RPC 2.0 spec
+                    message: "The request can't be parsed: invalid block height!".into(),
+                    data: None,
+                };
+                serde_json::to_value(&err).unwrap_or_default()
+            }
+        },
+        "getrawtransaction" => {
+            let resp = rpc
+                .get_raw_transaction(params[0].as_str().unwrap_or("").into())
+                .map_err(convert_crate_err);
+            serde_json::to_value(resp).unwrap_or_default()
+        }
+        "gettransactioninfo" => {
+            let resp = rpc
+                .get_transaction_info(params[0].as_str().unwrap_or("").into())
+                .map_err(convert_crate_err);
+            serde_json::to_value(resp).unwrap_or_default()
+        }
+        "decoderawtransaction" => {
+            let resp = rpc
+                .decode_raw_transaction(params[0].as_str().unwrap_or("").into())
+                .map_err(convert_crate_err);
+            serde_json::to_value(resp).unwrap_or_default()
+        }
+        "sendtransaction" => {
```

Comment:
Some naming is inconsistent (in the spec). 
```suggestion
        "sendrawtransaction" => {
```

Reply:
That's fair; let's leave this open for later in case we want to alter the name of the endpoint so that it matches the naming convention of the other `raw` ones.